lib/clients/AliClient.py

```diff
@@ -1,31 +1,115 @@
+        self.__aliCredentials = {
+            'access_key_id': access_key_id,
+            'secret_access_key': secret_access_key,
+            'region_name': region_name
+        }
+
+    def create_compute_client(self):
+        try:
+            credentials = self.__aliCredentials
+            compute_client = AcsClient(credentials['access_key_id'], credentials['secret_access_key'], credentials['region_name'], auto_retry=True,
+            max_retry_time=10, timeout=30)
+            return compute_client
+        except Exception as error:
+            raise Exception(
+                'Creation of compute client failed: {}'.format(error))
+
+    def create_storage_client(self):
+        try:
+            credentials = self.__aliCredentials
+            storage_client = oss2.Auth(
+                credentials['access_key_id'], credentials['secret_access_key'])
+            return storage_client
+        except Exception as error:
+            raise Exception(
+                'Creation of storage client failed: {}'.format(error))
+
+    def _get_common_request(self, action_name, params, tags=None):
+        request = CommonRequest()
+        request.set_domain('ecs.aliyuncs.com')
+        request.set_version('2014-05-26')
+        request.set_action_name(action_name)
+        i = 1
+        if tags != None:
+            for key in tags:
+                paramKey = 'Tag.' + str(i) + '.Key'
+                request.add_query_param(paramKey, key)
+                paramVal = 'Tag.' + str(i) + '.Value'
+                request.add_query_param(paramVal, tags[key])
+                i += 1
+        for key in params:
+            request.add_query_param(key, params[key])
+        return request
+
+    def _get_availability_zone_of_server(self, instance_id):
+        try:
+            instance_details_req_params = {
+                'InstanceIds' : [instance_id]
+            }
+            instance_details_request = self._get_common_request('DescribeInstances', instance_details_req_params)
+            instance_details = self.compute_client.do_action_with_exception(instance_details_request)
+            instance_details_json = json.loads(instance_details.decode('utf-8'))
+            if len(instance_details_json['Instances']['Instance']) > 1:
+                message = 'More than 1 instance found for with id {}'.format(
+                instance_id)
+                raise Exception(message)
+            return instance_details_json['Instances']['Instance'][0]['ZoneId']
+        except Exception as error:
+            self.logger.error(
+                '[ALI] ERROR: Unable to determine the availability zone of instance {}.\n{}'.format(instance_id, error))
+            return None
 
-    def get_container(self, auth, endpoint):
+    def get_container(self):
         try:
-            container = oss2.Bucket(auth, endpoint, self.CONTAINER)
+            container = oss2.Bucket(self.storage_client, self.endpoint, self.CONTAINER)
```

Comment:
Does `oss2` internally retries? I saw there wasn't any retries related parameters [here](https://github.com/cloudfoundry-incubator/service-fabrik-backup-restore/pull/57/files#diff-fe5e9623c538d6846d10874ca9426104R65)

Reply:
I couldn't find any option to pass retry param as such